oe_webtools_analytics/src/Event/AnalyticsEvent.php

```diff
@@ -0,0 +1,277 @@
+   *
+   * @var \Drupal\oe_webtools_analytics\Entity\SearchParametersInterface
+   */
+  private $search;
+
+  /**
+   * The analytic parameter.
+   *
+   * @var string
+   *   A string which by default it sets to "piwiki".
+   */
+  private $utility;
+
+  /**
+   * AnalyticsEvent constructor.
+   */
+  public function __construct() {
+    $this->search = new SearchParameters();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setSiteId(string $siteId): void {
+    $this->siteId = $siteId;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setSitePath(array $sitePath): void {
+    $this->sitePath = $sitePath;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setSiteSection(string $siteSection): void {
+    $this->siteSection = $siteSection;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setIs404Page(bool $is404Page = TRUE): void {
+    $this->is404Page = $is404Page;
+  }
+
+  /**
+   * Sets to true on 403 page.
+   *
+   * @param bool $is403Page
+   *   A boolean variable set as false by default.
+   */
+  public function setIs403Page(bool $is403Page = TRUE): void {
+    $this->is403Page = $is403Page;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setLangCode(string $langCode): void {
+    $this->langCode = $langCode;
+  }
+
+  /**
+   * Sets Instance to send the tracking information to different servers.
+   *
+   * Servers:
+   *    https://webanalytics.ec.europa.eu
+   *    https://webanalytics.europa.eu
+   *    https://webgate.ec.europa.eu/fpfis/piwik.
+   *
+   * @param string $instance
+   *   An optional string with "unknown" as default value.
+   */
+  public function setInstance(string $instance): void {
+    $this->instance = $instance;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setUtility(string $utility = 'piwik'): void {
+    $this->utility = $utility;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getSiteId(): string {
+    return $this->siteId;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getSiteSection(): string {
+    return $this->siteSection;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getSitePath(): array {
+    return $this->sitePath;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function is404Page(): bool {
+    return $this->is404Page;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function is403Page(): bool {
+    return $this->is403Page;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getLangCode(): string {
+    return $this->langCode;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getInstance(): string {
+    return $this->instance;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getSearch(): SearchParametersInterface {
+    return $this->search;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getUtility(): string {
+    return $this->utility;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function jsonSerialize() {
+    $data = [
```

Comment:
`get_object_vars($this)` might help here ?

Reply:
No, the members class are private!